libkbfs/cache/cache.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package cache
+
+import (
+	"sync"
+
+	"github.com/golang/groupcache/lru"
+)
+
+// Cache defines an interface for a cache that stores Measurable content.
+type Cache interface {
+	// Get tries to find and return data assiciated with key.
+	Get(key string) (data Measurable, ok bool)
+	// Add adds data into the cache, associating it with key. Entries are
+	// evicted when necessary.
+	Add(key string, data Measurable)
+}
+
+type randomEvictedCache struct {
+	maxBytes int
+
+	mu          sync.RWMutex
+	cachedBytes int
+	data        map[string]memorizedMeasurable
+}
+
+// NewRandomEvictedCache returns a Cache that uses random eviction strategy.
+// The cache will have a capacity of maxBytes bytes. A zero-byte capacity cache
+// is valid.
+//
+// Internally we store a memorizing wrapper for the raw Mesurable to avoid
```

Comment:
Both this and the LRU cache have an implicit assumption that the size of the object isn't going to change after it's cached.  That's not mentioned anywhere, here or in the `Measurable` interface definition, which does mention that the item can have "dynamic" parts, which sounds to me like it can be changed after its cached.  I know that's not what you meant, but it's not obvious and should probably be documented somewhere.

One option is to get rid of `Measurable` and just have the caller to `Add` pass in a size, which should make it clear that the size shouldn't change after the `Add`.  But if you want to keep the current way, I think clearer documentation is needed.